FirebaseCore/Sources/FIRVersion.m

```diff
@@ -14,20 +14,29 @@
  * limitations under the License.
  */
 
+#import "FirebaseCore/Sources/Private/FIRVersion.h"
+
 #ifndef Firebase_VERSION
 #error "Firebase_VERSION is not defined: add -DFirebase_VERSION=... to the build invocation"
 #endif
 
-#ifndef FIRCore_VERSION
-#error "FIRCore_VERSION is not defined: add -DFIRCore_VERSION=... to the build invocation"
-#endif
-
 // The following two macros supply the incantation so that the C
 // preprocessor does not try to parse the version as a floating
 // point number. See
 // https://www.guyrutenberg.com/2008/12/20/expanding-macros-into-string-constants-in-c/
 #define STR(x) STR_EXPAND(x)
 #define STR_EXPAND(x) #x
 
-const char *const FIRVersionString = (const char *const)STR(Firebase_VERSION);
-const char *const FIRCoreVersionString = (const char *const)STR(FIRCore_VERSION);
+@implementation FIRVersion
+
++ (NSString *)version {
+  const char *const FIRVersionString = (const char *const)STR(Firebase_VERSION);
+  static NSString *v;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    v = [NSString stringWithUTF8String:FIRVersionString];
+  });
+  return v;
```

Comment:
Do you think this is necessary vs instantiating the string every time? I can't imagine this is expensive to instantiate each time in the grand scheme of things.